src/lib/acpitbl.c

```diff
@@ -163,21 +182,24 @@ acpitbl_build_rdsp(void) {
 		0x00, 0x00, 0x00, 0x00
 	};
 
-	rdsp = (void *) (((uintptr_t) tb) + 0);
+	rdsp = (char *)tb + 0;
 	/* copy RDSP template to guest memory */
 	memcpy(rdsp, rdsp_tmpl, 36);
 	/* fixup table */
-	acpitbl_write32(rdsp, 0x10, ((uint32_t) (XHYVE_ACPI_BASE + RSDT_OFFSET)));
-	acpitbl_write64(rdsp, 0x18, ((uint64_t) (XHYVE_ACPI_BASE + XSDT_OFFSET)));
+	acpitbl_write32(rdsp, 0x10, (uint32_t)XHYVE_ACPI_BASE + RSDT_OFFSET);
```

Comment:
BTW, I would be fine with a "lets apply as is now and reconsider later" answer here...